subcommands/targets/offline-update.go

```diff
@@ -180,6 +182,15 @@ Notice that multiple targets in the same directory is only supported in LmP >= v
 			fmt.Printf("Downloading an ostree repo from the Target's OE build %d...\n", ti.ostreeVersion)
 			subcommands.DieNotNil(downloadOstree(factory, ti.ostreeVersion, ti.hardwareID, dstDir), "Failed to download Target's ostree repo:")
 		}
+
+		if ti.sha256 != nil {
+			sha256String := base64.StdEncoding.EncodeToString(ti.sha256)
+			expectedCommit := path.Join(dstDir, "ostree_repo", "objects", sha256String[0:2], sha256String[2:]+".commit")
+			if _, err := os.Stat(expectedCommit); errors.Is(err, os.ErrNotExist) {
+				fmt.Printf("WARNING: ostree repo does not contain target's hash %s, installation will fail. If the target references a custom ostree repo, re-run specifying --ostree-repo-source.\n", sha256String)
```

Comment:
IMO, this is the critical error, it doesn't make sense to ship bundle to devices if the commit hash is missing, so I would just exit/die at this spot.

Reply:
Changed to ERROR, and exiting